packages/jaeger-ui/src/utils/withRouteProps.tsx

```diff
@@ -0,0 +1,37 @@
+// Copyright (c) 2017 Uber Technologies, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+import React from 'react';
+import { useLocation, useParams } from 'react-router-dom';
+import { useHistory } from './useHistory';
+
+export default function withRouteProps(WrappedComponent: any) {
```

Comment:
I seem to recall we have a linter rule against `any`, does this PR increase the number of warnings?

Reply:
Yes actually. It increased. Fixed it in the latest commit.

Reply:
I just confirmed with the internal implementation of `withRouter`, and it also used `any` type, because we cannot predict the type of props that would be passed to the HOC.

```ts
export function withRouter<P extends RouteComponentProps<any>>(component: React.ComponentType<P>): React.ComponentClass<Omit<P, keyof RouteComponentProps<any>>>;
```